client/app/visit-history/visit-history.component.ts

```diff
@@ -1,74 +1,58 @@
 import { Component, OnInit } from '@angular/core';
-import Visit from '../shared/visit';
-import Volunteer from '../shared/volunteer';
+import { Volunteer } from '../shared/volunteer';
+import { testVisits, Visit } from '../shared/visit';
+import * as moment from 'moment-timezone';
 
 @Component({
   selector: 'app-visit-history',
   templateUrl: './visit-history.component.html',
   styleUrls: ['./visit-history.component.css']
 })
 export class VisitHistoryComponent implements OnInit {
-  public visits: Visit[];
+  public visitsByDate: Map<string, Visit[]>;
+  public dates: string[];
 
   constructor() { }
 
   ngOnInit() {
-    this.visits = this.getVisits();
+    this.visitsByDate = this.mapVisitsToDate(this.getVisits());
+    this.dates = Array.from(this.visitsByDate.keys());
   }
 
   /**
    * Use mock data until API is built
    * @returns {[{volunteer: Volunteer, startedAt: Date, endedAt: Date}]}
    */
   getVisits(): Visit[] {
-    const testVolunteer: Volunteer = {
-      firstName: 'Ted',
-      lastName: 'Mader',
-      petName: 'Mimi'
-    };
-    return [
-      {
-        volunteer: testVolunteer,
-        startedAt: new Date('2017-06-29T12:45:02.336Z'),
-        endedAt: new Date('2017-06-29T18:45:02.336Z')
-      },
-      {
-        volunteer: testVolunteer,
-        startedAt: new Date('2017-06-29T12:45:02.336Z'),
-        endedAt: new Date('2017-06-29T18:45:02.336Z')
-      },
-      {
-        volunteer: testVolunteer,
-        startedAt: new Date('2017-06-29T12:45:02.336Z'),
-        endedAt: new Date('2017-06-29T18:45:02.336Z')
-      }
-    ];
+    return testVisits;
   }
 
-  calculateDuration(visit: Visit) {
-    return visit.endedAt.getTime() - visit.startedAt.getTime();
+  mapVisitsToDate(visits: Visit[]) {
+    const map = new Map<string, Visit[]>();
+    visits.forEach(visit => {
+      const date = visit.startedAt.toDateString();
+      if (!map.has(date)) {
+        map.set(date, []);
+      }
+      map.get(date).push(visit);
+    });
+    return map;
   }
 
-  formatDate(date: Date): string {
-    return date.toDateString();
+  formatTime(date: Date, timezone: string): string {
+    const now = moment(date.toString());
+    return now.tz(timezone).format('h:m a');
   }
 
-  formatDuration(duration: number): string {
+  formatDuration(visit: Visit): string {
+    const duration = visit.endedAt.getTime() - visit.startedAt.getTime();
```

Comment:
I meant consider using the moment library to do your calculations instead of computing the duration yourself.